tests/integration/SdkTest_test.cpp

```diff
@@ -4696,6 +4704,117 @@ TEST_F(SdkTest, SdkGetBanners)
+        int err = synchronousSetMyBackupsFolder(0, mh);
+        ASSERT_TRUE(err == MegaError::API_OK) << "Setting handle for My Backup folder failed (error: " << err << ")";
+        // read the attribute to make sure it was set
+        mApi[0].h = 0; // reset this to test its value later
+        err = synchronousGetMyBackupsFolder(0);
+        ASSERT_TRUE(err == MegaError::API_OK);
+        ASSERT_EQ(mh, mApi[0].h) << "Getting handle for My Backup folder failed";
+    }
+
+    // look for Device Name attr
+    string deviceName;
+    if (synchronousGetDeviceName(0) == MegaError::API_OK && !attributeValue.empty())
+    {
+        deviceName = attributeValue;
+    }
+    else
+    {
+        auto now = m_time(nullptr);
+        char timebuf[32];
+        strftime(timebuf, sizeof timebuf, "%c", localtime(&now));
+
+        deviceName = string("Jenkins ") + timebuf;
+        synchronousSetDeviceName(0, deviceName.c_str());
+
+        // make sure Device Name attr was set
+        int err = synchronousGetDeviceName(0);
+        ASSERT_TRUE(err == MegaError::API_OK) << "Getting device name attr failed (error: " << err << ")";
+        ASSERT_EQ(deviceName, attributeValue) << "Getting device name attr failed (wrong value)";
+    }
+
+    // request to backup a folder
+    string folderToBackup = string(LOCAL_TEST_FOLDER) + "\\LocalBackedUpFolder";
+    makeNewTestRoot(folderToBackup.c_str());
+    mApi[0].h = 0;
+    const char* remoteBackup = "RemoteBackupFolder";
+    int err = synchronousBackupFolder(0, folderToBackup.c_str(), remoteBackup);
+    ASSERT_TRUE(err == MegaError::API_OK) << "Backup folder failed (error: " << err << ")";
+
+    // verify node attribute
+    std::unique_ptr<MegaNode> backupNode(megaApi[0]->getNodeByHandle(mApi[0].h));
+    const char* deviceIdFromNode = backupNode->getDeviceId();
+    std::unique_ptr<const char[]> deviceIdFromApi{ megaApi[0]->getDeviceId() };
+    ASSERT_STREQ(deviceIdFromNode, deviceIdFromApi.get());
+
+    // Verify that the remote path was created as expected
+    unique_ptr<char[]> myBackupsFolder{ megaApi[0]->getNodePathByNodeHandle(mh) };
+    string expectedRemotePath = string(myBackupsFolder.get()) + '/' + deviceName + '/' + remoteBackup;
+    unique_ptr<char[]> actualRemotePath{ megaApi[0]->getNodePathByNodeHandle(mApi[0].h) };
+    ASSERT_EQ(expectedRemotePath, actualRemotePath.get()) << "Wrong remote path for backup";
+
+    // Verify that the sync was added
+    unique_ptr<MegaSyncList> allSyncs{ megaApi[0]->getSyncs() };
+    ASSERT_TRUE(allSyncs && allSyncs->size()) << "API reports 0 Sync instances";
+    MegaSync* megaSync = allSyncs->get(0);
+    ASSERT_EQ(string(remoteBackup), megaSync->getName()) << "Sync instance points to a backup with wrong name";
+    ASSERT_EQ(string(actualRemotePath.get()), megaSync->getMegaFolder()) << "Sync instance points to wrong remote path";
+    ASSERT_EQ(mApi[0].h, megaSync->getMegaHandle()) << "Sync instance points to wrong MegaHandle";
```

Comment:
Instead of assuming that your sync will be the only one (today perhaps that's guaranteed, but not necessarily in the future), I think it would be better to iterate through the list, and find the one/s that have `MegaSync::getType() == TYPE_BACKUP`, then check for the name/path/handle has you do. If no sync of type backup matches, then fail.

Reply:
Updated.
I thought that tests had a guaranteed initial state, which included _no syncs_.

Reply:
Usually that's true, tests start with a clean state. However, the cache of configured syncs persist from one session to another. So in case a test fails/crashes and leaves leftovers, it may happen that some other syncs are restored from cache, potentially leading to a fake alarm on your test.
Thanks for the update